core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -829,7 +829,18 @@ private[spark] object Utils extends Logging {
   }
 
   private def getOrCreateLocalRootDirsImpl(conf: SparkConf): Array[String] = {
```

Comment:
Note that, this seems only being called via `Utils.getOrCreateLocalRootDirs` which caches `localRootDirs`. So, I think this won't mess up logs.